src/vue/forms/IssuanceForm.vue

```diff
@@ -0,0 +1,272 @@
+            {{ availableAmount.currency }}
+          </p>
+        </div>
+        <p
+          v-if="availableAmount.value"
+          class="issuance-form__available-amount-hint"
+        >
+          {{
+            'issuance.available-for-issuance-hint'
+              | globalize({ value: availableAmount })
+          }}
+        </p>
+      </div>
+    </div>
+    <div class="app__form-row">
+      <div class="app__form-field">
+        <input-field
+          white-autofill
+          v-model="form.receiver"
+          @blur="touchField('form.receiver')"
+          id="issuance-receiver"
+          :label="'issuance.receiver-lbl' | globalize"
+          :error-message="getFieldErrorMessage('form.receiver')"
+        />
+      </div>
+    </div>
+    <div class="app__form-row">
+      <div class="app__form-field">
+        <input-field
+          white-autofill
+          v-model="form.reference"
+          @blur="touchField('form.reference')"
+          id="issuance-reference"
+          :error-message="getFieldErrorMessage('form.reference')"
+          :label="'issuance.reference-lbl' | globalize"
+        />
+      </div>
+    </div>
+    <div class="app__form-actions">
+      <button
+        v-ripple
+        type="submit"
+        class="issuance-form__submit-btn"
+        :disabled="formMixin.isDisabled"
+      >
+        {{ 'issuance.issue-btn' | globalize }}
+      </button>
+      <button
+        v-ripple
+        type="button"
+        class="issuance-form__cancel-btn"
+        :disabled="formMixin.isDisabled"
+        @click.prevent="emitCancel"
+      >
+        {{ 'issuance.cancel-btn' | globalize }}
+      </button>
+    </div>
+  </form>
+</template>
+
+<script>
+import AssetLoaderMixin from '@/vue/mixins/asset-loader.mixin'
+import FormMixin from '@/vue/mixins/form.mixin'
+
+import { Bus } from '@/js/helpers/event-bus'
+import { ErrorHandler } from '@/js/helpers/error-handler'
+
+import { Sdk } from '@/sdk'
+import { base } from '@tokend/js-sdk'
+
+import { required, amountRange, emailOrAccountId } from '@validators'
+
+const EVENTS = {
+  cancel: 'cancel'
+}
+
+export default {
+  name: 'issuance-form',
+  mixins: [AssetLoaderMixin, FormMixin],
+  data: _ => ({
+    form: {
+      asset: '',
+      amount: 0,
+      receiver: '',
+      reference: ''
+    }
+  }),
+  validations () {
+    return {
+      form: {
+        asset: { required },
+        amount: {
+          required,
+          amountRange: amountRange(
+            0.000001,
+            this.availableAmount ? this.availableAmount.value : 0
+          )
+        },
+        receiver: { required, emailOrAccountId },
+        reference: { required }
+      }
+    }
+  },
+  computed: {
+    assetListValues () {
+      return this.ownedAssets
+        .map(asset => ({
+          value: asset.code,
+          label: `${asset.details.name} (${asset.code})`
+        }))
+    },
+    availableAmount () {
+      if (this.ownedAssets.length && this.form.asset) {
```

Comment:
if we don't have assets we own and can issue, the whole form makes no sense and can be replaced with some stub message

Reply:
so the whole logic you write here is actual only if `ownedAssets.length > 0`

Reply:
Resolved, updated

Reply:
Is there any possible case when you (or Vue) call the `availableAmount` property and `ownedAssets.length === 0`?